portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,54 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/message")
+@WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> comments;
+
+  @Override
+  public void init() {
+    comments = new ArrayList<>();
+    comments.add(
+        "A ship in port is safe, but that is not what ships are for. "
+            + "Sail out to sea and do new things. - Grace Hopper");
+    comments.add("They told me computers could only do arithmetic. - Grace Hopper");
+    comments.add("A ship in port is safe, but that's not what ships are built for. - Grace Hopper");
+    comments.add("It is much easier to apologise than it is to get permission. - Grace Hopper");
+    comments.add("If you can't give me poetry, can't you give me poetical science? - Ada Lovelace");
+    comments.add("I am in a charming state of confusion. - Ada Lovelace");
+    comments.add(
+        "The Analytical Engine weaves algebraic patterns, "
+            + "just as the Jacquard loom weaves flowers and leaves. - Ada Lovelace");
+    comments.add(
+        "Sometimes it is the people no one can imagine anything of "
+            + "who do the things no one can imagine. - Alan Turing");
+    comments.add("Those who can imagine anything, can create the impossible. - Alan Turing");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Indhu!");
+    String json = convertToJsonUsingGson(comments);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
   }
+
+  /**
+  * Converts an ArrayList instance into a JSON string using the Gson library.
+  */
+  private String convertToJsonUsingGson(List comments) {
+      Gson gson = new Gson();
```

Comment:
If you make the `Gson` object an instance variable, this method is no longer really needed.

initialize:
`private final Gson gson = new Gson();`

later...
`String json = gson.toJson(comments);`

Reply:
Thanks! I changed to this but I cannot make gson final because I am setting it.